pubsub.go

```diff
@@ -683,19 +683,15 @@ func (p *PubSub) handleDeadPeers() {
 
 		close(ch)
 
-		if p.host.Network().Connectedness(pid) == network.Connected {
+		if p.host.Network().Connectedness(pid) == network.Connected &&
+			!p.deadPeerBackoff.peerExceededBackoffThreshold(pid) {
```

Comment:
we might want to (debug) log this.

Reply:
https://github.com/libp2p/go-libp2p-pubsub/pull/483/commits/8b649669f077268770b6efd85333ff55053e5b97